src/neo/SmartContract/Manifest/ContractAbi.cs

```diff
@@ -44,11 +45,22 @@ public static ContractAbi FromJson(JObject json)
             };
         }
 
-        public ContractMethodDescriptor GetMethod(string name)
+        public ContractMethodDescriptor GetMethod(string name, int pcount)
         {
-            methodDictionary ??= Methods.ToDictionary(p => p.Name);
-            methodDictionary.TryGetValue(name, out var method);
-            return method;
+            if (pcount < -1 || pcount > ushort.MaxValue) throw new ArgumentOutOfRangeException(nameof(pcount));
+            methodDictionary ??= Methods.ToDictionary(p => (p.Name, p.Parameters.Length));
+            if (pcount >= 0)
+            {
+                methodDictionary.TryGetValue((name, pcount), out var method);
```

Comment:
I never tried with a key like this, i am not sure if it's faster a dictionary of dictionaries

Reply:
I think they are the same.